src/ray/raylet/worker_pool.cc

```diff
@@ -1470,7 +1468,37 @@ void WorkerPool::PrestartWorkers(const TaskSpecification &task_spec,
                    << backlog_size << " and available CPUs " << num_available_cpus
                    << " num idle workers " << state.idle.size()
                    << " num registered workers " << state.registered_workers.size();
-    PrestartDefaultCpuWorkers(task_spec.GetLanguage(), num_needed);
+    PrestartWorkersWithRuntimeEnv(task_spec, num_needed);
+  }
+}
+
+void WorkerPool::PrestartWorkersWithRuntimeEnv(const TaskSpecification &task_spec,
+                                               int64_t num_needed) {
+  RAY_LOG(DEBUG) << "PrestartWorkers " << num_needed;
+  for (int ii = 0; ii < num_needed; ++ii) {
+    GetOrCreateRuntimeEnv(
```

Comment:
Need this `if (!IsRuntimeEnvEmpty(serialized_runtime_env)) {`

Reply:
The whole function is for with-runtime case?